File: ios/Components/RNModule/DBManagerModule/DBManagerModule.m
//
//  DBManagerModule.m
//  RNPorn
//
//  Created by Nemo on 2018/12/13.
//  Copyright © 2018年 Facebook. All rights reserved.
//

#import "DBManagerModule.h"
#import "DBHelper.h"

@implementation DBManagerModule

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(createDBTableName:(NSString *)name params:(NSArray *)params) {
  [[DBHelper sharedInstance] createDBParams:params table:name];
}

RCT_EXPORT_METHOD(complyTableName:(NSString *)name SQL:(NSString *)sql res:(RCTResponseSenderBlock)callback) {
  id result = [[DBHelper sharedInstance] complySQL:sql tableName:name];
  callback(@[result]);
}

RCT_EXPORT_METHOD(addDataTableName:(NSString *)name dict:(NSDictionary *)dict ){
  [[DBHelper sharedInstance] addDataWithParam:dict table:name];
}

RCT_EXPORT_METHOD(addMoreDataTableName:(NSString *)name array:(NSArray *)array  callBack:(RCTResponseSenderBlock)callback){
  BOOL b = [[DBHelper sharedInstance] addMoreDataWithParam:array table:name];
  callback(@[[NSNumber numberWithBool:b]]);
}

RCT_EXPORT_METHOD(getAllDataTableName:(NSString *)name callBack:(RCTResponseSenderBlock)callback) {
  NSArray *datas = [[DBHelper sharedInstance] getAllDataWithTableName:name];
  callback(@[datas]);
}

RCT_EXPORT_METHOD(getSingleDataTableName:(NSString *)name key:(NSString *)key value:(NSString *)value callBack:(RCTResponseSenderBlock)callback) {
  NSArray *datas = [[DBHelper sharedInstance] getSingleDataWithTableName:name key:key value:value];
  callback(@[datas]);
}

RCT_EXPORT_METHOD(deleteSingleDataTableName:(NSString *)tabName condition:(NSString *)condition allBack:(RCTResponseSenderBlock)callback) {
  BOOL b = [[DBHelper sharedInstance] deleteSingleData:condition tableName:tabName];
  callback(@[[NSNumber numberWithBool:b]]);
}

RCT_EXPORT_METHOD(deleteAllDataTableName:(NSString *)tabName) {
  [[DBHelper sharedInstance] deleteAllDataTableName:tabName];
}

@end

File: android/app/src/main/java/com/rnporn/Components/RNModule/DBManagerModule.java
package com.rnporn.Components.RNModule;


import android.util.Log;

import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.WritableArray;
import com.rnporn.Components.DB.DBHelper;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.bridge.Callback;

public class DBManagerModule extends ReactContextBaseJavaModule {

    private ReactContext mReactContext;

    public DBManagerModule(ReactApplicationContext reactContext) {
        super(reactContext);
        mReactContext = reactContext;
    }

    @Override
    public String getName() {
        return "DBManagerModule";
    }

    @ReactMethod
    public void addDataTableName(String tableName,ReadableMap map) {
        DBHelper.getInstance().add(tableName,map);
    }

    @ReactMethod
    public void addMoreDataTableName(String tableName,ReadableArray array,Callback callback) {
        callback.invoke(DBHelper.getInstance().batchAdd(tableName,array));
    }

    @ReactMethod
    public void createDBTableName(String name, ReadableArray params) {
        DBHelper dbManager =  DBHelper.getInstance();
        dbManager.rnReactContext = mReactContext;
        dbManager.createDB(name,params);
    }

    @ReactMethod
    public void getAllDataTableName(String name,Callback callback) {
        callback.invoke(DBHelper.getInstance().getAllData(name));
    }

    @ReactMethod
    public void getSingleDataTableName(String name,String key,String value,Callback callback) {
        callback.invoke(DBHelper.getInstance().getSingleData(name,key,value));
    }

    @ReactMethod
    public void deleteSingleDataTableName(String name, String condition,Callback callback) {
        DBHelper.getInstance().deleteSingleData(name,condition);
        callback.invoke(true);
    }

    @ReactMethod
    public void deleteAllDataTableName(String name) {
        DBHelper.getInstance().deleteAllData(name);
    }
}
